lib/stripe/stripe_client.rb

```diff
@@ -64,9 +80,9 @@ def self.default_client
     end
 
     # A default connection manager for the current thread.
-    def self.default_connection_manager
+    def self.default_connection_manager(config = Stripe.configuration)
       current_thread_context.default_connection_manager ||= begin
-        connection_manager = ConnectionManager.new
+        connection_manager = ConnectionManager.new(config)
```

Comment:
Aye, that's a great catch. I completely missed `||=` there.  

I'm in favor of option two. Seems like it would better satisfy the functionality that's requested in https://github.com/stripe/stripe-ruby/pull/966.

Reply:
@brandur-stripe I've implemented the second approach and added a couple of specs to capture any regressions for the bug.

At first, I used the configuration's `object_id` to key the connection managers, but realized that could potentially increase the memory footprint when configuration objects are identical and aren't eligible for manual GC. I decided to opt for a deterministic key here: https://github.com/stripe/stripe-ruby/blob/190052b580722e8d62a87c9f406b6e6ae8c0d98e/lib/stripe/stripe_configuration.rb#L188-L193

An alternative would be to use `Marshal.dump`, but I'm a bit skeptical about the performance (I haven't benchmarked, but I'm more than happy to.)